tools/privacyidea-migrate-linotp.py

```diff
@@ -173,182 +220,135 @@
-        desc = r["Description"]
-        type = r["Type"]
-
-        if name not in resolvers:
-            resolvers[name] = {"type": "ldapresolver",
-                               "config": [{"Key": "CACHE_TIMEOUT",
-                                           "Value": "120"},
-                                          {"Key": "SCOPE",
-                                           "Value": "SUBTREE",
-                                           "Type": "string"},
-                                          {"Key": "EDITABLE",
-                                           "Value": "0",
-                                           "Type": "bool"}]}
-        resolvers[name]["config"].append({
-            "Key": resolver_key,
-            "Value": value,
-            "Description": desc,
-            "Type": type
-        })
-
-if resolvers:
-    # Write the resolvers
-    for name, config in resolvers.iteritems():
-        print("Migrating Resolver {0!s}".format(name))
-
-        ins = resolver_table.insert().values(
-                name=name,
-                rtype=config.get("type")
-        )
-        res_id = conn_pi.execute(ins)
-        resolver_id = res_id.inserted_primary_key[0]
-        for resolver_entry in config.get("config"):
-            # save the list of resolver configuration
-            ins = resolver_config_table.insert().values(
-                resolver_id=resolver_id,
-                Key=resolver_entry.get("Key") or "",
-                Value=resolver_entry.get("Value") or "",
-                Type=resolver_entry.get("Type") or "",
-                Description=resolver_entry.get("Description") or ""
-            )
-            conn_pi.execute(ins)
-
-
-# Put resolvers into realms
-s = select([linotp_config_table])
-result = conn_linotp.execute(s)
-realms = {}
-for r in result:
-    config_key = r["Key"]
-    entry = config_key.split(".")
-    if entry[1] == "useridresolver" and entry[2] == "group":
-        resolvers = r["Value"].split()
-        realms[entry[3]] = [x.split(".")[3] for x in resolvers]
-
-print("Migrating realm definitions {0}".format(realms))
-
-# Save the resolvers in the realms
-"""For this we need to get the IDs of the realmname and the resolvernames"""
-for realmname, resolvers in realms.iteritems():
-    print("Migrating realm {0!s}".format(realmname))
-    s = select([realm_table]).where(realm_table.c.name == realmname)
+    s = select([resolver_table])
     result = conn_pi.execute(s)
     for r in result:
-        realm_id = r["id"]
+        resolver_id_map[r["name"]] = r["id"]
+
+    print("Realm-Map: {}".format(realm_id_map))
+    print("Resolver-Map: {}".format(resolver_id_map))
+
+# Process Tokens
+
+if MIGRATE.get("tokens"):
+    s = select([linotp_token_table])
+    result = conn_linotp.execute(s)
 
-    for resolvername in resolvers:
-        s = select([resolver_table]).where(resolver_table.c.name ==
-                                           resolvername)
+    i = 0
+    for r in result:
```

Comment:
I'm a fan of
``for i, r in enumerate(result, 1): ...``
but I think that's not that important for a helper script :)

Reply:
I do not understand this :-/

Reply:
Instead of using an explicit counter variable ``i``, one can use ``enumerate``:
```python
>>> for i, letter in enumerate(['a', 'b', 'c', 'd'], 1): print i, letter
... 
1 a
2 b
3 c
4 d
```
... but this does not seem that important to me :)